pom.xml

```diff
@@ -285,4 +292,79 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <profiles>
+        <profile>
+            <id>test</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <skipGpgSign>true</skipGpgSign>
+            </properties>
+        </profile>
+        <!-- Deployment profile (required so these plugins are only used when deploying) -->
+        <profile>
+            <id>deploy</id>
+            <properties>
```

Comment:
one other thing: i think we should disable the `maven-shade-plugin` and `nexus-staging-maven-plugin` plugins for the deployment profile.

Reply:
i'm not totally sure how this `<profile>` section interplays with the `<plugins>` section above though, are the above build plugins not used if we're using `deploy`?

Reply:
It looks like it will run both the plugins from the profile and the generally defined plugins. [Run before](https://app.circleci.com/pipelines/github/nightfallai/nightfall-java-sdk/191/workflows/b629625d-25d5-4976-a82f-7ff1e0da6e6e/jobs/183) https://github.com/nightfallai/nightfall-java-sdk/pull/87/commits/d20da29ec11487b625d2e8203e661901e9d02432 [run after](https://app.circleci.com/pipelines/github/nightfallai/nightfall-java-sdk/192/workflows/b6c75f22-b505-49ad-9575-19332cc2b42e/jobs/184). They both run the same plugins, its just that the after run is much more verbose in the logs.

Reply:
Do we need all of those general plugins to run for the deploy? If not we can move them into the test profile.

Reply:
i'm pretty certain we need the license one in the deploy profile. not 100% sure what the "release" one does off the top of my head, but sounds like we probably need it in deploy rather than test. the "checkstyle" one is fine for testing only, probably ditto on that "staging" one. and i think this deploy section is also missing the "jar" plugin from above, that one is probably needed for both?

Reply:
Updated so that there is not a general build plugins section. Everything is now either under the test or deploy profile